decidim-core/lib/decidim/map/provider/static_map/here.rb

```diff
@@ -6,8 +6,42 @@ module Provider
       module StaticMap
         # The static map utility class for the HERE maps service
         class Here < ::Decidim::Map::StaticMap
+          def url(latitude:, longitude:, options: {})
+            map_url = configuration.fetch(:url, nil)
+            return super unless map_url
+
+            return super unless map_url.include?("mia/v3")
+
+            w = options[:width] || Decidim::Map::StaticMap::DEFAULT_SIZE
+            h = options[:height] || Decidim::Map::StaticMap::DEFAULT_SIZE
+
+            params = {
+              apiKey: configuration[:api_key],
+              overlay: "point:#{latitude},#{longitude};icon=cp;size=large|#{latitude},#{longitude};style=circle;width=50m;color=%231B9D2C60"
+            }
+
+            URI.parse("#{map_url}:radius=90/#{w}x#{h}/png8").tap do |uri|
+              uri.query = URI.encode_www_form(params)
+            end.to_s
+          end
+
           # @See Decidim::Map::StaticMap#url_params
           def url_params(latitude:, longitude:, options: {})
+            ActiveSupport::Deprecation.warn(
```

Comment:
Shouldn't we actually check that the old format is used before showing this warning?

Reply:
Actually no, because I have overridden the `url` from `Decidim::Map::StaticMap`. The `url` method from `Here` class is not using the `url_params`. You may want to check lines 11 & 13 from this class

https://github.com/decidim/decidim/blob/e20ae0423bd6af5b71975b55fd29018bd52857e2/decidim-core/lib/decidim/map/provider/static_map/here.rb#L10-L13

Reply:
Ah I didn't catch that one. Then it is great!